### PR TITLE
Create ndjbdns 1.0.6 formula

### PR DESCRIPTION
**Pull request by deployable**
The original djbdns is not really safe to install any more. There are number of security issues and patches  required to bring it up to date and the ndjbdns project includes these. http://pjp.dgplug.org/ndjbdns/ https://github.com/pjps/ndjbdns/

This does conflict with the djbdns formula though as it's basically the same thing.

### CODE REVIEW

File: Library/Formula/ndjbdns.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
+class Ndjbdns < Formula
+  desc "N-DJBDNS is a brand new release of DJBDNS."
```
Comment (dunn) [inline Library/Formula/ndjbdns.rb]: This makes it sound like a version bump, rather than a fork.

File: Library/Formula/ndjbdns.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    inreplace "#{etc}/ndjbdns/walldns.conf",  /^([UG]ID)=\d+/, '\1=1'
+  end
+
+  test do
+    # would need to run a server process in the background
+    #system "#{bin}/dnscache -P 1053"
+    system "dnsip localhost"
+  end
+
+  conflicts_with 'djbdns',
```
Comment (dunn) [inline Library/Formula/ndjbdns.rb]: This should be above the `depends_on` lines.

File: Library/Formula/ndjbdns.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    :because => 'both install the same binaries'
+
+end
+
+
+# Patch while waiting upstream to fix: https://github.com/pjps/ndjbdns/issues/17
```
Comment (dunn) [inline Library/Formula/ndjbdns.rb]: Having this comment above `patch :DATA` is good enough.

File: Library/Formula/ndjbdns.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
+class Ndjbdns < Formula
+  desc "N-DJBDNS is a brand new release of DJBDNS."
+  homepage "http://pjp.dgplug.org/djbdns/"
+  url "http://pjp.dgplug.org/djbdns/ndjbdns-1.06.tar.gz"
+  sha256 "5ce5a7c5031f220a85fc8bca903f2d3cf484ff77e4c85e7144a0e2a5922a1127"
+  
+  depends_on "autoconf"
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: These should either be `=> :build` or `=> :run` dependencies. It looks like the former.

File: Library/Formula/ndjbdns.rb [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+  # Patch while waiting upstream to fix: https://github.com/pjps/ndjbdns/issues/17
+  patch :DATA
+
+  def install
+    bin.mkpath
+    (prefix+"etc").mkpath
+    # disable the chance of init.d ./etc setup
+    inreplace "etc/Makefile.am", "SUBDIRS = ip init.d servers", "SUBDIRS = ip servers"
+    system "autoreconf -if"
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: Split your arguments. `system "autoreconf", "-if"`

File: Library/Formula/ndjbdns.rb [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    system "./configure --prefix=#{prefix} --sysconfdir=#{etc}"
+    system "make"
+    system "make install"
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: And here: `system "make", "install"`

File: Library/Formula/ndjbdns.rb [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  # Patch while waiting upstream to fix: https://github.com/pjps/ndjbdns/issues/17
+  patch :DATA
+
+  def install
+    bin.mkpath
+    (prefix+"etc").mkpath
+    # disable the chance of init.d ./etc setup
+    inreplace "etc/Makefile.am", "SUBDIRS = ip init.d servers", "SUBDIRS = ip servers"
+    system "autoreconf -if"
+    system "./configure --prefix=#{prefix} --sysconfdir=#{etc}"
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: And here: `system "./configure", "--prefix=#{prefix}", "--sysconfdir=#{etc}"`

File: Library/Formula/ndjbdns.rb [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    system "make"
+    system "make install"
+    inreplace "#{etc}/ndjbdns/axfrdns.conf",  /^([UG]ID)=\d+/, '\1=1'
+    inreplace "#{etc}/ndjbdns/dnscache.conf", /^([UG]ID)=\d+/, '\1=1'
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: For multiple inreplaces for the same file you should use the:

```ruby
inreplace "ducks.txt" do |s|
  s.gsub! "ducks", "horses"
  s.gsub! "ducks2", "horses2"
end
```
syntax.

File: Library/Formula/ndjbdns.rb [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    inreplace "#{etc}/ndjbdns/dnscache.conf", /IP=.+/, 'IP=0.0.0.0'
+    inreplace "#{etc}/ndjbdns/rbldns.conf",   /^([UG]ID)=\d+/, '\1=1'
+    inreplace "#{etc}/ndjbdns/tinydns.conf",  /^([UG]ID)=\d+/, '\1=1'
+    inreplace "#{etc}/ndjbdns/walldns.conf",  /^([UG]ID)=\d+/, '\1=1'
+  end
+
+  test do
+    # would need to run a server process in the background
+    #system "#{bin}/dnscache -P 1053"
+    system "dnsip localhost"
```
Comment (DomT4) [inline Library/Formula/ndjbdns.rb]: You need to call the directory on this, and split the arguments again.

**Comment by DomT4** [comment]
Please run `brew audit --strict --online /path/to/formula.rb` on new submissions. Thanks. There's a lot here that would have been highlighted.

File: Library/Formula/ndjbdns.rb (hunk 1 above)
Comment (deployable) [inline Library/Formula/ndjbdns.rb]: It's their lead line from http://pjp.dgplug.org/ndjbdns/, I've added fork in there.

**Comment by deployable** [comment]
That commit should cover the comments so far

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
Install is failing:
```
./configure: line 4689: syntax error near unexpected token `systemd,'
./configure: line 4689: `PKG_CHECK_EXISTS(systemd, have_systemd=yes, have_systemd=no)'
```
http://bot.brew.sh/job/Homebrew%20Pull%20Requests/38266/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_ndjbdns/

**Comment by DomT4** [comment]
Closing as dead. Please feel free to resubmit with comments addressed if desired. Thank for your submission to Homebrew!

**Pull request closed without merging**